homeassistant/components/openuv/__init__.py

```diff
@@ -47,6 +51,36 @@
 
 PLATFORMS = [Platform.BINARY_SENSOR, Platform.SENSOR]
 
+SERVICE_NAME_UPDATE_DATA = "update_data"
+SERVICE_NAME_UPDATE_PROTECTION_DATA = "update_protection_data"
+SERVICE_NAME_UPDATE_UV_INDEX_DATA = "update_uv_index_data"
+
+SERVICES = (
+    SERVICE_NAME_UPDATE_DATA,
+    SERVICE_NAME_UPDATE_PROTECTION_DATA,
+    SERVICE_NAME_UPDATE_UV_INDEX_DATA,
+)
+
+SERVICE_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_DEVICE_ID): cv.string,
+    }
+)
+
+
+@callback
+def async_get_openuv_for_service_call(hass: HomeAssistant, call: ServiceCall) -> OpenUV:
+    """Get the OpenUV object related to a service call (by device ID)."""
+    device_id = call.data[CONF_DEVICE_ID]
+    device_registry = dr.async_get(hass)
+
+    if device_entry := device_registry.async_get(device_id):
+        for entry in hass.config_entries.async_entries(DOMAIN):
```

Comment:
instead of going through all OpenUV entries, it is probably faster to go through the config entries attached to the device. The first one to belong to OpenUV is the one (and probably the first)